File: app/src/test/java/com/mgaetan89/showsrage/model/LogLevel_ToStringTest.java
package com.mgaetan89.showsrage.model;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class LogLevel_ToStringTest {
	@Parameterized.Parameter(0)
	public LogLevel logLevel;

	@Parameterized.Parameter(1)
	public String result;

	@Test
	public void toStringTest() {
		assertThat(this.logLevel.toString()).isEqualTo(this.result);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{LogLevel.DEBUG, "debug"},
				{LogLevel.ERROR, "error"},
				{LogLevel.INFO, "info"},
				{LogLevel.WARNING, "warning"},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/model/LogLevel_ToStringTest.kt
package com.mgaetan89.showsrage.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class LogLevel_ToStringTest(val logLevel: LogLevel, val result: String) {
	@Test
	fun toStringTest() {
		assertThat(this.logLevel.toString()).isEqualTo(this.result)
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data(): Collection<Array<Any>> {
			return listOf(
					arrayOf(LogLevel.DEBUG, "debug"),
					arrayOf(LogLevel.ERROR, "error"),
					arrayOf(LogLevel.INFO, "info"),
					arrayOf(LogLevel.WARNING, "warning")
			)
		}
	}
}
